lib/features/azure_live_log.js

```diff
@@ -52,8 +52,8 @@ export default class LiveLog {
     var queue = task.runtime.queue;
 
     // Create date when this artifact should expire (see config).
-    var expiration =
-      new Date(Date.now() + task.runtime.logging.liveLogExpires);
+    var expiration = new Date(Math.min(Date.now() + task.runtime.logging.liveLogExpires,
+      task.task.expires ? new Date(task.task.expires) : Infinity));
```

Comment:
When would `task.expires` be missing? We get the task from `queue.task` response schema says that it's always present...
The queue inserts a default if it's missing :)